client/src/sagas/apartments.js

```diff
@@ -0,0 +1,47 @@
+import { takeLatest, call, put} from 'redux-saga/effects';
+import NodeGeocoder from 'geocoder';
+
+import { APARTMENTS_REQUEST, APARTMENTS_SUCCESS, APARTMENTS_FAILURE } from '../constants/apartments';
+
+import fetchApartmentsApi from '../api/apartments';
+
+// Used to map the response of the API request for the apartments
+const mapApartmentsResponse = (apartmentRecords) => {
+  const markers = [];
+  return apartmentRecords.map(record => {
+    return NodeGeocoder.geocode(record.street_address, (err, geoCodedAddress) => {
+      const location = geoCodedAddress.results[0].geometry.location
+      return markers.push({
+        key: record.id,
+        position: {
+          lat: location.lat,
+          lng: location.lng
+        },
+        defaultAnimation: 2
+      });
+    });
+  });
+};
+
+function* apartmentsFlow(action) {
```

Comment:
It took me a really long time to understand how generators work, especially with redux saga. However, I'd say they're designed to make async actions a lot simpler, so they definitely shouldn't overcomplicate things. The biggest thing with using redux saga is using their methods for invoking and receiving data from async actions (e.g. using `call` for making an API request and using `put` for calling an action to be used in a reducer).

My friend sent me [this](http://start.jcolemorrison.com/react-and-redux-sagas-authentication-app-tutorial/) tutorial for learning how to create an auth flow with redux saga, and it definitely helped clear stuff up for me. My implementation is pretty similar to the [basic implementation](https://github.com/redux-saga/redux-saga#sagasjs) in the docs.